File: frontEnd/src/types/dto/base/StepBaseDTO.ts
import {AssertStepDTO} from "@/types/dto/stepDTO/AssertStepDTO";
import {ExtractStepDTO} from "@/types/dto/stepDTO/ExtractStepDTO";
import {Setting} from "@/types/dto/SettingDTO";
import {StepType} from "@/types/enum/StepType";

export class StepBaseDTO {
    parentId: number = 0;// 父节点id
    stepName: string = '';// 步骤名称

    stepType: StepType | string = '';// 步骤类型
    assertList?: AssertStepDTO[] = []; // 断言列表
    extractList?: ExtractStepDTO[] = []; // 提取列表
    setting?: Setting = new Setting();// 步骤设置
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/StepBaseDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;

import com.mokatest.platform.demos.domain.ui.Setting;
import lombok.Data;

import java.util.List;

@Data
public abstract class StepBaseDTO {
    // 步骤名称
    private String stepName;

//    // 步骤类型
//    private StepType stepType;

    // 断言列表
    private List<AssertStepDTO> assertList;

    // 提取列表
    private List<ExtractStepDTO> extractList;

    // 步骤设置
    private Setting setting;

}